ide/src/focus.ts

```diff
@@ -25,130 +27,224 @@ interface FocusState {
   disposable: vscode.Disposable;
 }
 
```

Comment:
This class is now sufficiently complicated / stateful that it can use a doc-comment explaining it's lifecycle. Like it would be useful to say what the states are, and how the user's interactions change the states and call specific functions.